### ISSUE TITLE
Implement shallow-ensemble based uncertainty quantification

### ISSUE THREAD

**Issue by Benzoin96485** [Benzoin96485/Enerzyme#58]
As proposed in [Mach. Learn.: Sci. Technol. 5 035006](https://iopscience.iop.org/article/10.1088/2632-2153/ad594a/meta), shallow ensemble is promising way of quantifying the uncertainty of a neural network potential accurately and efficiently.

In this issue, we propose to implement
- Shallow ensemble for the last layer of PhysNet and SpookyNet, and all the associated modifications in downstream layers and losses regarding to the tensor shapes
- A shallow ensemble reduce layer for the calculation of the shallow ensemble's mean and variation statistics
- Active learning cycle using shallow ensemble's output
- The NLL loss and CRPS loss proposed in the paper
- Monitor extending to shallow ensemble statistics
- Post-training re-evaluation for the test set partition, with interested intermediate results outside the target list
